hazelcast/src/main/java/com/hazelcast/client/impl/protocol/util/Flyweight.java

```diff
@@ -97,4 +83,22 @@ protected void uint32Put(final int offset, final long value, final ByteOrder byt
 
     //endregion PUT/GET helpers
 
+    protected void ensureCapacity(final int requiredCapacity) {
+        final int capacity = buffer.capacity();
+        if (requiredCapacity > capacity) {
+            final int newCapacity = findSuitableCapacity(capacity, requiredCapacity);
+            ByteBuffer newBuffer =  ByteBuffer.allocate(newCapacity);
+            //TODO if not using byte buffer ????
+            newBuffer.put(buffer.byteBuffer());
+            buffer.wrap(newBuffer);
+        }
+    }
+    private static int findSuitableCapacity(int capacity, final int requiredCapacity) {
+        do {
+            capacity <<= 1;
+        } while (capacity < requiredCapacity);
+
+        return capacity;
```

Comment:
@ahmetmircik, see method `ensureCapacity()`, it checks for the capacity first and then calls `findSuitableCapacity()` if required.

```java
if (requiredCapacity > capacity) {
   final int newCapacity = findSuitableCapacity(capacity, requiredCapacity);
   ....
}
```